agent/consul/leader_connect.go

```diff
@@ -224,10 +224,21 @@ func (s *Server) initializeRootCA(provider ca.Provider, conf *structs.CAConfigur
 		return fmt.Errorf("error getting intermediate cert: %v", err)
 	}
 
+	// Check if the configured leaf cert TTL would be less than allowed by the provider.
+	// If so, abort with an error. Also, grab the configured PK type for later use when
+	// creating CSRs.
 	commonConfig, err := conf.GetCommonConfig()
 	if err != nil {
 		return err
 	}
+	if commonConfig.LeafCertTTL.Seconds() == 0 {
+		commonConfig.LeafCertTTL = provider.MinimumLeafTTL()
+		s.logger.Printf("[WARN] leaf TTL not configured, setting to provider minimum of %v",
+			provider.MinimumLeafTTL())
+	} else if commonConfig.LeafCertTTL < provider.MinimumLeafTTL() {
+		return fmt.Errorf("configured leaf TTL of %v is less than provider minimum of %v",
```

Comment:
What happens if this fails? Could we just bump the leaf TTL up so that it matches the provider one?

Reply:
According to the RFC, we should fail to configure the CA, unless the `ForceWithoutCrossSigning` parameter has been set to `true`. That will be implemented in a following PR.